### PR TITLE
Add initial SecondaryDeposits tests

### PR DESCRIPTION
**Pull request by i-stam**

### CODE REVIEW

File: test/contracts/SecondaryDeposit.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+				name: 'Mock Token',
+				symbol: 'MCK',
+				supply: mockTokenTotalSupply,
+			}));
+
+			helper.incrementExpectedBalance(account1, mockTokenTotalSupply);
+			await this.token.transfer(account1, 100, { from: owner });
+		});
+
+		it('has the expected parameters', async () => {
```
Comment (justinjmoses) [inline test/contracts/SecondaryDeposit.js:49]: heads up that you don't have to check mock token params - it's a mock not a test target

File: test/contracts/SecondaryDeposit.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+			assert.equal('18', await this.token.decimals());
+			assert.equal(toWei(mockTokenTotalSupply), await this.token.totalSupply());
+			assert.equal(100, await this.token.balanceOf(account1));
+		});
+
+		describe('when a mock for Issuer is added', () => {
+			beforeEach(async () => {
+				mock = await artifacts.require('GenericMock').new();
```
Comment (justinjmoses) [inline test/contracts/SecondaryDeposit.js:57]: `mockIssuer`?

File: test/contracts/SecondaryDeposit.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
+const { artifacts, contract, web3 } = require('@nomiclabs/buidler');
```
Comment (justinjmoses) [inline test/contracts/SecondaryDeposit.js:1]: Please name this file `.unit.js`

File: contracts/SecondaryDeposit.sol [hunk 4]
```diff
@@ -140,7 +141,8 @@ contract SecondaryDeposit is Owned, MixinResolver, MixinSystemSettings, ISeconda
         uint /*amount*/
     ) external {
         revert("Not implemented");
-        // ensure function only callable from SecondaryDeposit2 (via messenger)
+        // ensure function only callable from SecondaryDeposit2 via messenger (aka relayer)
+        require(msg.sender == address(messenger()), "Only the relayer can call this");
```
Comment (justinjmoses) [inline contracts/SecondaryDeposit.sol:145]: this can be commented out for now.

File: test/contracts/SecondaryDeposit.unit.js [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+		describe('when mocks for Issuer and resolver are added', () => {
+			beforeEach('deploy a mock issuer contract', async () => {
+				this.issuerMock = await artifacts.require('GenericMock').new();
+
+				// now instruct the mock Issuer that debtBalanceOf() must return 0
+				await mockGenericContractFnc({
+					instance: this.issuerMock,
+					mock: 'Issuer',
+					fncName: 'debtBalanceOf',
```
Comment (justinjmoses) [inline test/contracts/SecondaryDeposit.unit.js:62]: we should also ensure that it reverts when this is non-zero

File: test/contracts/SecondaryDeposit.unit.js [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+						this.issuerMock.address,
+						companion,
+						{
+							from: deployerAccount,
+						}
+					);
+				});
+
+				it('has the expected parameters', async () => {
+					assert.bnEqual(await this.secondaryDeposit.maximumDeposit(), maxDeposit);
```
Comment (justinjmoses) [inline test/contracts/SecondaryDeposit.unit.js:107]: we should also test it reverts if the amount is greater

File: test/contracts/SecondaryDeposit.unit.js [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+					});
+					it('should emit a MintedSecondary event', async () => {
+						assert.eventEqual(mintSecondaryTx, 'MintedSecondary', {
+							account: account1,
+							amount: 100,
+						});
+					});
+				});
+
+				describe('when called by the owner', async () => {
```
Comment (justinjmoses) [inline test/contracts/SecondaryDeposit.unit.js:154]: when what is called?

File: test/contracts/SecondaryDeposit.unit.js (hunk 5 above)
Comment (i-stam) [reply test/contracts/SecondaryDeposit.unit.js:62]: i pushed those tests as well.

File: test/contracts/SecondaryDeposit.unit.js (hunk 6 above)
Comment (i-stam) [reply test/contracts/SecondaryDeposit.unit.js:107]: same as above

**Pull request merged**